R/conditions.R

```diff
@@ -435,37 +439,91 @@ cnd_body.vctrs_error_recycle_incompatible_size <- function(cnd, ...) {
 
 # Names -------------------------------------------------------------------
 
-stop_names <- function(message, class, locations, ...) {
-  stop_vctrs(
-    message,
-    class = c(class, "vctrs_error_names"),
-    locations = locations,
-    ...
+new_error_names <- function(class = NULL, names) {
+  new_error_vctrs(
+    c(class, "vctrs_error_names"),
+    names = names
   )
 }
 
-stop_names_cannot_be_empty <- function(locations) {
-  stop_names(
-    "Names must not be empty.",
+stop_names_cannot_be_empty <- function(names) {
+  cnd_signal(new_error_names(
```

Comment:
I think we can keep using `abort()` and wrappers here? I only used `error_cnd()` in the subscript code to avoid throwing and make it easier and faster to compose errors, but I don't think this should be the default approach.